client/gutenberg/editor/controller.js

```diff
@@ -88,6 +90,21 @@ export const loadTranslations = store => {
 	} );
 };
 
+export const redirect = async ( context, next ) => {
+	let state = context.store.getState();
+	let siteId = getSelectedSiteId( state );
+	if ( ! siteId ) {
+		siteId = await waitForSelectedSiteId( context );
+		state = context.store.getState();
+	}
+
+	if ( ! isGutenbergEnabled( state, siteId ) ) {
+		page.redirect( `/post/${ getSelectedSiteSlug( state ) }` );
+	}
```

Comment:
note that @jsnajdr [pointed out](
https://github.com/Automattic/wp-calypso/pull/29166#discussion_r239374366) a bug that was fixed in #28449 so that we can rely on having a selected site in memory by the time this middleware is called. that means we don't need to wait for it.

```js
export const redirect = ( { store: { getState } }, next ) => {
	const state = getState();
	const siteId = getSelectedSiteId( state );
	const hasGutenberg = isGutenbergEnabled( state, siteId );

	if ( hasGutenberg ) {
		return next();
	}

	return page.redirect( `/post/${ getSelectedSiteSlug( state ) }` );
};
```

Reply:
What @dmsnell said. Calypso framework recently got a tiny bit better, letting section authors worry less about weird states of site selection.

Reply:
> Calypso framework recently got a tiny bit better, letting section authors worry less about weird states of site selection.

Oooooohh, fantastic ❤️